sklearn/neighbors/base.py

```diff
@@ -258,6 +258,19 @@ def _fit(self, X):
         else:
             raise ValueError("algorithm = '%s' not recognized"
                              % self.algorithm)
+
+        if self.n_neighbors is not None:
```

Comment:
Some classes like radiusneighbors which inherits neighborbase do not need ```n_neighbors```. ```n_neighbors``` could be ```None```, I think. @banilo